frontend/src/metabase/visualizations/components/settings/ChartSettingHideButton.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useCallback } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { t } from "ttag";
+import cloneDeep from "lodash.clonedeep";
+import _ from "underscore";
+
+import Button from "metabase/core/components/Button";
+import { updateCardVisualizationSettings } from "metabase/query_builder/actions";
+import type { Column } from "metabase-types/types/Dataset";
+import type { State } from "metabase-types/store";
+
+import { TableColumnOrderSetting } from "metabase-types/api";
+import { normalizeFieldRef } from "metabase-lib/queries/utils/dataset";
+
+interface ChartSettingHideButtonProps {
+  column: Column;
+  columns: Array<Column>;
+  onClose?: () => void;
+}
+
+export function ChartSettingHideButton({
+  column,
+  columns,
+  onClose,
+}: ChartSettingHideButtonProps) {
+  const hideColumn = useHideColumn(column, columns);
+
+  return (
+    <Button
+      onlyText
+      icon="eye_crossed_out"
+      onClick={() => {
+        hideColumn();
+        onClose?.();
+      }}
+    >
+      {t`Hide this column`}
+    </Button>
+  );
+}
+
+function useHideColumn(column: Column, columns: Array<Column>) {
+  const dispatch = useDispatch();
+  const visualizationSettings = useSelector(
+    (state: State) => state.qb.card?.visualization_settings,
+  );
+
+  return useCallback(() => {
+    let visSettingsClone = cloneDeep(visualizationSettings);
+    if (visSettingsClone === undefined) {
+      visSettingsClone = {};
+    }
+    if (visSettingsClone["table.columns"] === undefined) {
+      visSettingsClone["table.columns"] = getDefaultTableColumns(columns);
+    }
+
+    const tableColIndex = findTableColumnIndex(
+      visSettingsClone?.["table.columns"],
+      column,
+    );
+    if (tableColIndex === -1) {
+      return;
+    }
+
+    const tableCol = visSettingsClone["table.columns"][tableColIndex];
+    visSettingsClone["table.columns"][tableColIndex] = {
+      ...tableCol,
+      enabled: false,
+    };
+    // @ts-expect-error thunk
```

Comment:
The `dispatch` method returned by `useDispatch` by default does not expect a function as it's argument. There is a workaround by defining your own custom dispatch type (see [stackoverflow thread](https://stackoverflow.com/questions/59800913/type-safe-usedispatch-with-redux-thunk)). 

I think we should re-export `useDispatch` with the proper type to avoid this error whenever we need to dispatch a thunk, but for better separation of concerns I plan to do that in a followup PR and fixing this then.